src/huggingface_hub/file_download.py

```diff
@@ -484,15 +484,22 @@ def http_get(
     hf_raise_for_status(r)
     content_length = r.headers.get("Content-Length")
     total = resume_size + int(content_length) if content_length is not None else None
+
+    displayed_name = url
+    content_disposition = r.headers.get("Content-Disposition")
+    if content_disposition is not None and "filename=" in content_disposition:
+        # Means file is on CDN
+        displayed_name = content_disposition.split("filename=")[-1]
```

Comment:
Not related to download speed. I've added that myself to fix the progress bar naming :roll_eyes: